### DISCUSSION TITLE
Custom withDefault function

### DISCUSSION THREAD

**Discussion by sillvva** [open-circle/valibot#111] [General]
Wrote a custom `withDefault` function, because the existing one didn't work for my needs.

This one converts falsy values or (in the case of numbers) NaN values to the default value.

```typescript
import type { BaseSchema, BaseSchemaAsync, Input, ValidateInfo } from "valibot";

export function withDefault<TSchema extends BaseSchema | BaseSchemaAsync>(schema: TSchema, value: Input<TSchema>) {
	return {
		...schema,
		parse(input: Input<TSchema>, info?: ValidateInfo) {
			return schema.parse(!input || (typeof value == "number" && isNaN(input)) ? value : input, info);
		}
	};
}
```

Leaving this here, in case anyone else finds it useful.

**Comment by fabian-hiller** [comment]
Yes, the fact that you can easily write your own methods and schemas is the cool thing about Valibot. 🧩 `withDefault` is currently only used explicitly for `undefined` values. Since we are revising `withDefault` at the moment, I will take a look at that again.

Since the default value is only applied to `undefined`, `withDefault` works with any schema. However, I am aware that there are use cases where you might want to use a default value also in the case of `null` or a falsy value, for example.
